commands/account/auth.ts

```diff
@@ -0,0 +1,212 @@
+import { setAsDefaultAccountPrompt } from '../../lib/prompts/setAsDefaultAccountPrompt';
+import { EXIT_CODES } from '../../lib/enums/exitCodes';
+import { uiFeatureHighlight } from '../../lib/ui';
+import { CommonArgs, ConfigArgs } from '../../types/Yargs';
+
+const i18nKey = 'commands.account.subcommands.auth';
+
+const TRACKING_STATUS = {
+  STARTED: 'started',
+  ERROR: 'error',
+  COMPLETE: 'complete',
+};
+
+async function createOrUpdateConfig(
+  env: Environment,
+  doesConfigExist: boolean,
+  disableTracking: boolean | undefined,
+  authType: string,
+  account?: number
+): Promise<CLIAccount | null> {
+  try {
+    const { personalAccessKey } = await personalAccessKeyPrompt({
+      env,
+      account,
+    });
+    const token = await getAccessToken(personalAccessKey, env);
+    const defaultName = token.hubName ? toKebabCase(token.hubName) : undefined;
+
+    const name = doesConfigExist
+      ? undefined
+      : (await cliAccountNamePrompt(defaultName)).name;
+
+    const updatedConfig = await updateConfigWithAccessToken(
+      token,
+      personalAccessKey,
+      env,
+      name,
+      !doesConfigExist
+    );
+
+    if (!updatedConfig) return null;
+
+    if (doesConfigExist && !updatedConfig.name) {
+      updatedConfig.name = (await cliAccountNamePrompt(defaultName)).name;
+      updateAccountConfig({
+        ...updatedConfig,
+      });
+      writeConfig();
+    }
+
+    return updatedConfig;
+  } catch (e) {
+    if (!disableTracking) {
+      await trackAuthAction('account-auth', authType, TRACKING_STATUS.ERROR);
+    }
+    debugError(e);
+    return null;
+  }
+}
+
+export const describe = undefined; // i18n(`${i18nKey}.describe`);
+export const command = 'auth';
+
+type AccountAuthArgs = CommonArgs &
+  ConfigArgs & {
+    disableTracking?: boolean;
+  };
+
+export async function handler(
+  args: ArgumentsCamelCase<AccountAuthArgs>
+): Promise<void> {
+  const { providedAccountId, disableTracking } = args;
+  const authType = PERSONAL_ACCESS_KEY_AUTH_METHOD.value;
+
+  const deprecatedConfigExists = configFileExists(false);
+  if (deprecatedConfigExists) {
+    logger.error(
+      i18n(`${i18nKey}.errors.bothConfigFilesNotAllowed`, {
+        deprecatedConfig: DEFAULT_HUBSPOT_CONFIG_YAML_FILE_NAME,
+      })
+    );
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  if (!disableTracking) {
+    trackCommandUsage('account-auth', {}, providedAccountId);
+    await trackAuthAction('account-auth', authType, TRACKING_STATUS.STARTED);
+  }
+
+  const env = args.qa ? ENVIRONMENTS.QA : ENVIRONMENTS.PROD;
+  const configExists = configFileExists(true);
+
+  if (!configExists) {
+    if (!disableTracking) {
+      await trackAuthAction('account-auth', authType, TRACKING_STATUS.STARTED);
+    }
+    createEmptyConfigFile({}, true);
+  }
+  loadConfig('');
+
+  handleExit(deleteEmptyConfigFile);
+
+  const updatedConfig = await createOrUpdateConfig(
+    env,
+    configExists,
```

Comment:
Ah, I think I see what's happening here. Is `configExists` used to tell `createOrUpdateConfig` if the config has newly been created? It might be a little clearer to name it `configIsEmpty` or something like that